### PR TITLE
chore(kgo): add changelog for 1.4.0

### PR DESCRIPTION
**Pull request by pmalek**
### Description

Extracted from #8083 

### Testing instructions

Preview link: https://deploy-preview-8101--kongdocs.netlify.app/gateway-operator/changelog/#140

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

**Comment by czeslavo** [comment]
https://github.com/Kong/docs.konghq.com/pull/8083 has been merged so I believe this can be closed.

**Comment by pmalek** [comment]
> #8083 has been merged so I believe this can be closed.

#8083 didn't include the changelog as that's unconditionally rendered and without it we were able to merge all the other bits for 1.4.0 . It was removed in https://github.com/Kong/docs.konghq.com/pull/8083/commits/e8bccdbac4a8ab92a7c4b461100358eec4bd8947

**Review by lena-larionova — APPROVED** [review]
Approving for whenever you're ready to release.

File: app/gateway-operator/changelog.md [hunk 1]
```diff
@@ -5,6 +5,120 @@ no_version: true
 
 Changelog for supported {{ site.kgo_product_name }} versions.
 
+## 1.4.0
+
+**Release Date** TBA
```
Comment (lena-larionova) [inline app/gateway-operator/changelog.md:10]:
```suggestion
**Release Date** 2024/10/31
```

**Pull request merged**